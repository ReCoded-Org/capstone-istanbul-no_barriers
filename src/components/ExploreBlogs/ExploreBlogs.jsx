import React from "react";

const ExploreBlogs = () => {

}

export default ExploreBlogs