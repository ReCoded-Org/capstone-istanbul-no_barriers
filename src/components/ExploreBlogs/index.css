/* Color Variables */
:root {
  --main-bg-color: #d7f7fa;
  --main-color: #00374d;
  --white-link: #eff4f6;
  --orange-link: #f2c94c;
  --dark-orange: #d88315;
}

.exploreBlogs {
  background-color: var(--main-bg-color);
  color: var(--main-color);
  padding: 4rem 4rem;
}

.exploreBlogsBlogList {
  padding: 0 5rem 1.5rem 5rem;
}

.exploreBlogsBlogImg {
  height: 17rem;
  width: auto;
  border-radius: 1rem;
  margin: 1rem 0;
}

.exploreBlogsBlogLink {
  text-decoration: none;
  color: var(--main-color);
}

.exploreBlogsBlogLink:hover {
  text-decoration: none;
  color: var(--orange-link);
}

/* button color on hover */
button.btn.btn-outline-warning:hover {
  color: var(--white-link);
}

/* button on click */
.btn-outline-warning:not(:disabled):not(.disabled):active {
  color: var(--white-link);
  background-color: var(--dark-orange);
  border: none;
}
