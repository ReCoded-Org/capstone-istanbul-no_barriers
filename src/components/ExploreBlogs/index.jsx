import React from "react";
import "./index.css";
import { Row, Container, Col, Button } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import image from "./images/placeHolderImage.jpg";

// Explore-Blogs section shows the last 3 blog posts in the database.
// If you click on a single blog post, you go to that blog's page
// If you click on show more button, you go to blogs list page
const ExploreBlogs = () => {
  const { t } = useTranslation();

  return (
    <div className="exploreBlogs">
      <Row className="justify-content-center">
        <Container className="exploreBlogsHeading">
          <h3 className="exploreBlogsHeadingTitleMain">
            {t("homePage.exploreBlogs.exploreBlogs")}
          </h3>
          <p className="exploreBlogsHeadingParagraph">
            {t("homePage.exploreBlogs.paragraph")}
          </p>
        </Container>
      </Row>
      <Container className="exploreBlogsBlogList">
        <Row className="d-flex">
          <Col xs={12} md={6} lg={4}>
            <div className="exploreBlogsBlog">
              <img
                src={image}
                alt="placeholder"
                className="exploreBlogsBlogImg"
              />
              <h5 className="exploreBlogsBlogHeading">
                {t("homePage.exploreBlogs.blogTitle")}
              </h5>
              <a href="#singleBlog" className="exploreBlogsBlogLink">
                {t("homePage.exploreBlogs.blogLink")}
              </a>
            </div>
          </Col>
          <Col xs={12} md={6} lg={4}>
            <div className="exploreBlogsBlog">
              <img
                src={image}
                alt="placeholder"
                className="exploreBlogsBlogImg"
              />
              <h5 className="exploreBlogsBlogHeading">
                {t("homePage.exploreBlogs.blogTitle")}
              </h5>
              <a href="#singleBlog" className="exploreBlogsBlogLink">
                {t("homePage.exploreBlogs.blogLink")}
              </a>
            </div>
          </Col>
          <Col xs={12} md={6} lg={4}>
            <div className="exploreBlogsBlog">
              <img
                src={image}
                alt="placeholder"
                className="exploreBlogsBlogImg"
              />
              <h5 className="exploreBlogsBlogHeading">
                {t("homePage.exploreBlogs.blogTitle")}
              </h5>
              <a href="#singleBlog" className="exploreBlogsBlogLink">
                {t("homePage.exploreBlogs.blogLink")}
              </a>
            </div>
          </Col>
        </Row>
      </Container>
      <Row className="justify-content-center">
        <Button variant="outline-warning shadow-none">
          {t("homePage.exploreBlogs.buttonSeeMore")}
        </Button>
      </Row>
    </div>
  );
};

export default ExploreBlogs;
