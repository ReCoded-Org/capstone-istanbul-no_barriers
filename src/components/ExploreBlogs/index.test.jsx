import React from "react";
import renderer from "react-test-renderer";
import ExploreBlogs from "./index";

it("Explore Blogs renders correctly", () => {
  const tree = renderer.create(<ExploreBlogs />).toJSON();
  expect(tree).toMatchSnapshot();
});
