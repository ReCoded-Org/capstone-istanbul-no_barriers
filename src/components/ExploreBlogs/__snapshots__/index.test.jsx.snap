// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explore Blogs renders correctly 1`] = `
<div
  className="exploreBlogs"
>
  <div
    className="justify-content-center row"
  >
    <div
      className="exploreBlogsHeading container"
    >
      <h3
        className="exploreBlogsHeadingTitleMain"
      >
        homePage.exploreBlogs.exploreBlogs
      </h3>
      <p
        className="exploreBlogsHeadingParagraph"
      >
        homePage.exploreBlogs.paragraph
      </p>
    </div>
  </div>
  <div
    className="exploreBlogsBlogList container"
  >
    <div
      className="d-flex row"
    >
      <div
        className="col-lg-4 col-md-6 col-12"
      >
        <div
          className="exploreBlogsBlog"
        >
          <img
            alt="placeholder"
            className="exploreBlogsBlogImg"
            src="placeHolderImage.jpg"
          />
          <h5
            className="exploreBlogsBlogHeading"
          >
            homePage.exploreBlogs.blogTitle
          </h5>
          <a
            className="exploreBlogsBlogLink"
            href="#singleBlog"
          >
            homePage.exploreBlogs.blogLink
          </a>
        </div>
      </div>
      <div
        className="col-lg-4 col-md-6 col-12"
      >
        <div
          className="exploreBlogsBlog"
        >
          <img
            alt="placeholder"
            className="exploreBlogsBlogImg"
            src="placeHolderImage.jpg"
          />
          <h5
            className="exploreBlogsBlogHeading"
          >
            homePage.exploreBlogs.blogTitle
          </h5>
          <a
            className="exploreBlogsBlogLink"
            href="#singleBlog"
          >
            homePage.exploreBlogs.blogLink
          </a>
        </div>
      </div>
      <div
        className="col-lg-4 col-md-6 col-12"
      >
        <div
          className="exploreBlogsBlog"
        >
          <img
            alt="placeholder"
            className="exploreBlogsBlogImg"
            src="placeHolderImage.jpg"
          />
          <h5
            className="exploreBlogsBlogHeading"
          >
            homePage.exploreBlogs.blogTitle
          </h5>
          <a
            className="exploreBlogsBlogLink"
            href="#singleBlog"
          >
            homePage.exploreBlogs.blogLink
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="justify-content-center row"
  >
    <button
      className="btn btn-outline-warning shadow-none"
      disabled={false}
      type="button"
    >
      homePage.exploreBlogs.buttonSeeMore
    </button>
  </div>
</div>
`;
